modules/core/exported/client.go

```diff
@@ -49,7 +49,8 @@ type ClientState interface {
 	// Clients must return their status. Only Active clients are allowed to process packets.
 	Status(ctx sdk.Context, clientStore sdk.KVStore, cdc codec.BinaryCodec) Status
 
-	// Checks for evidence of a misbehaviour in Header or Misbehaviour type
+	// Checks for evidence of a misbehaviour in Header or Misbehaviour type. It may assume the ClientMessage
```

Comment:
Since you say below in the godoc for `VerifyClientMessage` that _Calls to CheckForMisbehaviour, UpdateState, and UpdateStateOnMisbehaviour will assume that the content of the ClientMessage has been verified_, then maybe better:

```suggestion
	// Checks for evidence of a misbehaviour in Header or Misbehaviour type. It assumes the ClientMessage
```